### ISSUE TITLE
`coercePropertyValue` Does Not Account For Array of ObjectID or Array of ObjectID strings

### ISSUE THREAD

**Issue by taicho** [strongloop/loopback-connector-mongodb#545]
# Description/Steps to reproduce
Arrays of ObjectID or ObjectId-like-strings cannot be saved/updated/created and attempting to do so results in an error.

1. Create any Entity/Model.
2. Decorate a model w/ `{strictObjectIDCoercion:true}`.
2. Decorate a property with either:
```ts
@property.array(String, {mongodb: {dataType:'ObjectID'}})
```
OR
```ts
@property({
     type:'array',
     itemType: 'string',
     mongodb: {
        dataType: 'ObjectID',   
    }
})
```
I've tried many other variations. No matter what, `coercePropertyValue` in `mongodb.js` does not account for the fact that the `propValue` might be an Array *unless* it is a decimal128 array rather than an Array of ObjectID or ObjectID-like-strings.

# Expected result
The expectation is that one should be able to simply add an ObjectId string (or even ObjectId  - this results in an error as well) to the array and update/save/create the data.

# Notes
- Also attempting to persist a property with an empty array (that is decorated with any of the above property decorators) will result in an error due to coercePropertyValue continuing down the same code path and trying to evaluate the property value.
- I am mainly submitting this issue to make the owners aware of the problem and to hopefully aid a traveler in search of an answer or information related to this problem.

Tested on version: 5.0.1

**Comment by dhmlau** [comment]
Related to PR https://github.com/strongloop/loopback-connector-mongodb/pull/517

Cross posting @taicho's [comment in the PR](https://github.com/strongloop/loopback-connector-mongodb/pull/517#discussion_r334069205):

> Created #545 as I believe it relates to this conversation. This commit and its related code appears to fix the issue related to string-properties-stored-as-ObjectId but does not account for a property that is an Array of ObjectID or Array of ObjectID-like-strings.

**Comment by dhmlau** [comment]
There's an open PR #550. Reopening this issue.

**Comment by mrbatista** [comment]
@dhmlau related (merged) PR #560

**Comment by dhmlau** [comment]
Thanks @mrbatista. Closing this issue as done.
